### ISSUE TITLE
Stellarium 1.1: Images of objects like the moon, planets etc. disappear when zooming in

### ISSUE THREAD

**Issue by zeklh** [Stellarium/stellarium#2818]
When I zoom into the moon, planets or deep sky objects which are extended like galaxies they disappear at a certain zoom level. There is only the background to be seen. This is  independent whether I activate atmosphere or not.

**Comment by alex-w** [comment]
Please attach the log.txt from Stellarium and what about value of certain zoom level?

**Comment by zeklh** [comment]
Tested with the moon: the image starts to get dimmer at a field of view value of about 40 and has disappeared at about 26.

Here is the log-file:
[log.txt](https://github.com/Stellarium/stellarium/files/9927348/log.txt)

**Comment by 10110111** [comment]
Let me guess: by the image you mean a kind of halo, right? What if you wait a minute or two, does a ball appear?

I have noticed that the models of planets are only loaded after you zoom into them, and only after some time passes do they actually appear.

**Comment by 10110111** [comment]
Oh, actually your problem is more severe: the planets won't load because shader compilation failed.

**Comment by 10110111** [comment]
Please try this patch

```diff
diff --git a/data/shaders/planet.frag b/data/shaders/planet.frag
index d760c9e53f..0f49882e12 100644
--- a/data/shaders/planet.frag
+++ b/data/shaders/planet.frag
@@ -267,7 +267,7 @@ void main()
     normal = normalize(normalX*normal.x+normalY*normal.y+normalZ*normal.z);
     // normal now contains the real surface normal taking normal map into account
 
-	mediump float horizonShadowCoefficient = 1;
+	mediump float horizonShadowCoefficient = 1.;
 	{
 		// Check whether the fragment is in the shadow of surrounding mountains or the horizon
 		mediump vec3 lonDir = normalX;
@@ -279,39 +279,39 @@ void main()
 		mediump vec4 sinHorizElevs = sign(horizonElevSample) * horizonElevSample * horizonElevSample;
 		mediump float sinHorizElevLeft, sinHorizElevRight;
 		mediump float alpha;
-		if(sunAzimuth >= PI/2)
+		if(sunAzimuth >= PI/2.)
 		{
 			// Sun is between East and South
 			sinHorizElevLeft = sinHorizElevs[1];
 			sinHorizElevRight = sinHorizElevs[2];
-			alpha = (sunAzimuth - PI/2) / (PI/2);
+			alpha = (sunAzimuth - PI/2.) / (PI/2.);
 		}
-		else if(sunAzimuth >= 0)
+		else if(sunAzimuth >= 0.)
 		{
 			// Sun is between North and East
 			sinHorizElevLeft = sinHorizElevs[0];
 			sinHorizElevRight = sinHorizElevs[1];
-			alpha = sunAzimuth / (PI/2);
+			alpha = sunAzimuth / (PI/2.);
 		}
-		else if(sunAzimuth <= -PI/2)
+		else if(sunAzimuth <= -PI/2.)
 		{
 			// Sun is between South and West
 			sinHorizElevLeft = sinHorizElevs[2];
 			sinHorizElevRight = sinHorizElevs[3];
-			alpha = (sunAzimuth + PI) / (PI/2);
+			alpha = (sunAzimuth + PI) / (PI/2.);
 		}
 		else
 		{
 			// Sun is between West and North
 			sinHorizElevLeft = sinHorizElevs[3];
 			sinHorizElevRight = sinHorizElevs[0];
-			alpha = (sunAzimuth + PI/2) / (PI/2);
+			alpha = (sunAzimuth + PI/2.) / (PI/2.);
 		}
 		mediump float horizElevLeft = asin(sinHorizElevLeft);
 		mediump float horizElevRight = asin(sinHorizElevRight);
 		mediump float horizElev = horizElevLeft + (horizElevRight-horizElevLeft)*alpha;
 		if(sinSunElevation < sin(horizElev))
-			horizonShadowCoefficient = 0;
+			horizonShadowCoefficient = 0.;
 	}
 #else
     // important to normalize here again
```

**Comment by zeklh** [comment]
I think, this happens for every extended object where images have to be loaded. So, it's the loading process which does not work.

**Comment by zeklh** [comment]
How to activate the patch?

**Comment by 10110111** [comment]
Find `planet.frag` file in the Stellarium installation directory and replace with [this file](https://github.com/Stellarium/stellarium/blob/master/data/shaders/planet.frag). This should fix it.

**Comment by zeklh** [comment]
Works for solar system objects but nor for deep sky objects like M57 or M31.

**Comment by 10110111** [comment]
The problem with DSO needs a separate issue, because I don't reproduce it, while this one with planets is fixed.

**Comment by zeklh** [comment]
I just saw that the issue is closed. But there is still a problem with deep sky objects. Shall I open a new issue or will this be reopened?
